src/main/java/com/spartronics4915/frc2019/subsystems/PanelHandler.java

```diff
@@ -88,21 +92,33 @@ public void onLoop(double timestamp)
                 SystemState newState = defaultStateTransfer();
                 switch (mSystemState)
                 {
-                    case RETRACTING:
+                    case ARM_DOWNING:
+                        if (mStateChanged)
+                        {
+                            mArmSolenoid.set(Constants.kPanelSolenoidRetract);
+                        }
+                    case ARM_UPING:
                         if (mStateChanged)
                         {
-                            mSolenoid.set(Constants.kPanelSolenoidRetract);
+                            mArmSolenoid.set(Constants.kPanelSolenoidExtend);
                         }
-                        break;
-                    case EJECTING://BB6
+                    case EJECTING_ROCKET:
                         if (mStateChanged)
                         {
-                            mSolenoid.set(Constants.kPanelSolenoidExtend);
+                            mArmSolenoid.set(Constants.kPanelSolenoidRetract);
+                            mArmDownTime = Timer.getFPGATimestamp();
+                        }
+                        else if (Timer.getFPGATimestamp() > mArmDownTime + Constants.kPanelArmDownTime){
+                            mSlideSolenoid.set(Constants.kPanelSolenoidExtend);
                             mEjectTime = Timer.getFPGATimestamp();
+                        } 
+                        else if (Timer.getFPGATimestamp() > mEjectTime + Constants.kPanelEjectTime){
+                            mSlideSolenoid.set(Constants.kPanelSolenoidRetract);
+                            mRetractTime = Timer.getFPGATimestamp();
+                        }
+                        else if (Timer.getFPGATimestamp() > mRetractTime + Constants.kPanelRetractTime && newState == mSystemState && mWantedState == WantedState.EJECT_ROCKET){
```

Comment:
You don't need `newState == mSystemState` here, because you're not changing the system state.